Dockerfile

```diff
@@ -19,10 +16,17 @@ RUN  go mod download
 # Copy the go source
 COPY . $work_dir/
 # Build
-RUN  go build -a -o $work_dir/bin/controller $work_dir/services/$service_alias/cmd/controller/main.go
+RUN GIT_VERSION=$(git describe --tags --dirty --always) && \
+    GIT_COMMIT=$(git rev-parse HEAD) && \
+    BUILD_DATE=$(date +%Y-%m-%dT%H:%M) && \
+    CGO_ENABLED=0 GOOS=linux GOARCH=amd64 GO111MODULE=on \
+    go build -ldflags="-X ${VERSION_PKG}.GitVersion=${GIT_VERSION} \
+    -X ${VERSION_PKG}.GitCommit=${GIT_COMMIT} \
+    -X ${VERSION_PKG}.BuildDate=${BUILD_DATE}" \
+    -a -o $work_dir/bin/controller $work_dir/services/$service_alias/cmd/controller/main.go
```

Comment:
Is the reason you changed the above from using ENV variables to using CLI variables because you needed to execute shell commands to populate things like GIT_COMMIT? If so, why move the static env vars like CGO_ENABLED and GOOS to the command line invocation?

Reply:
I thought as these are just build an argument, so I went ahead and added them as CLI variables. I can revert this change and add all of these as env vars.